src/lua/api-gateway/aws/AwsV4Signature.lua

```diff
@@ -141,36 +145,44 @@ function HmacAuthV4Handler:formatQueryString(uri_args)
     return uri
 end
 
-function HmacAuthV4Handler:getSignature(http_method, request_uri, uri_arg_table, request_payload )
+function HmacAuthV4Handler:getSignature(http_method, request_uri, uri_arg_table, request_payload, host_override)
     local uri_args = self:formatQueryString(uri_arg_table)
     local utc = ngx.utctime()
     local date1 = self.aws_date_short
     local date2 = self.aws_date
+    local host = self.aws_endpoint
```

Comment:
When does `aws_endpoint` differ from `host` ? For S3 use-case does it include the full path ?